src/Kiota.Builder/CodeParameterOrderComparer.cs

```diff
@@ -3,17 +3,34 @@
 namespace Kiota.Builder;
 public class CodeParameterOrderComparer : IComparer<CodeParameter>
 {
+
+    private List<string> parameterTypeOrders = new ();
+
+    public CodeParameterOrderComparer(List<string> defaultParamOrder)
+    {
+        parameterTypeOrders = defaultParamOrder;
+    }
+    public CodeParameterOrderComparer()
+    {
+    }
+
     public int Compare(CodeParameter x, CodeParameter y)
     {
         return (x, y) switch {
             (null, null) => 0,
             (null, _) => -1,
             (_, null) => 1,
-            _ => x.Optional.CompareTo(y.Optional) * optionalWeight +
-                getKindOrderHint(x.Kind).CompareTo(getKindOrderHint(y.Kind)) * kindWeight +
-                x.Name.CompareTo(y.Name) * nameWeight,
+            _ => getDefaultOrder(y.Type).CompareTo(getDefaultOrder(x.Type)) * defaultParamOrderWeight +
+                 x.Optional.CompareTo(y.Optional) * optionalWeight +
+                 getKindOrderHint(x.Kind).CompareTo(getKindOrderHint(y.Kind)) * kindWeight +
+                 x.Name.CompareTo(y.Name) * nameWeight,
         };
     }
+    
+    private int getDefaultOrder(CodeTypeBase codeType)
+    {
+        return parameterTypeOrders.IndexOf(codeType.Name);
+    }
```

Comment:
```suggestion
```

No need for any of that, just edit the switch below.

Reply:
In this case I used `CodeParameterKind.Custom` for context param, changing the switch statement might result in changing the order of all custom parameters , the alternative will be to hardcode the parameter name which isn't better in this case

Reply:
aren't you using the cancellation token kind given the changes made in the refiner?

Reply:
Changed it to `CodeParameterKind.Custom` ,  I can revert it to Cancellation

Reply:
go ahead, we might rename the kind key to make it more generic in the future

Reply:
done